File: 2016/d1/sol1.ex
input = File.read!("input.txt")
  |> String.trim
  |> String.split(",", trim: true)

# [N, E, S, W]
dir = 0
x = 0
y = 0

Enum.each(input, fn(x) ->
  # Handle turning around
  case String.at(x, 0) do
    "L" ->
      dir = case dir do
        0 ->
          3
        true ->
          dir + 1
      end
    "R" ->
      dir = case dir do
        3 ->
          0
        true ->
          dir + 1
      end
    true -> "Error"
  end
  # Handle walking forward in the new direction.
  case dir do
    0 ->
      x = x + String.to_integer(String.slice(x, 1, -1))
    1 ->
      y = y + (String.slice(x, 1, -1) |> String.to_integer)
    2 ->
      x = x - (String.slice(x, 1, -1) |> String.to_integer)
    3 ->
      y = y - (String.slice(x, 1, -1) |> String.to_integer)
    true -> "Error"
  end
end )

IO.puts x <> ", " <> y

File: 2017/d12/sol1.rb
input = File.read("input.txt").split("\n")

remaining = *(0..1999)
groups = 0

while remaining.size > 0
  toAdd = [remaining[0]]
  groups = groups + 1
  while toAdd.size > 0
    newVals = input[toAdd.pop].split[2..-1].map(&:to_i)
    toAdd = toAdd + (newVals & remaining)
    remaining = remaining - newVals
  end
end

puts groups
